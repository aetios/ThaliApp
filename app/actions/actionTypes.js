export const NAVIGATE = 'NAVIGATE';
export const BACK = 'BACK';
export const OPENDRAWER = 'OPENDRAWER';
export const CALENDARRETREIVED = 'CALENDARRETREIVED';
export const CALENDARERROR = 'CALENDARERROR';
export const RESETLOGINSTATE = 'RESETLOGINSTATE';
export const WELCOME = 'WELCOME';
