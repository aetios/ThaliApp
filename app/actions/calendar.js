/**
 * Created by aucke on 3/8/17.
 */
