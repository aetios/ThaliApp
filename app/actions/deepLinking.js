export const DEEPLINK = 'DEEPLINKING_DEEPLINK';

export function deepLink(url, stayInApp = true) {
  return {
    type: DEEPLINK,
    payload: { url, stayInApp },
  };
}
