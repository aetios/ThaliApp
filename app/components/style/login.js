import { colors, create } from '../../style';

const styles = create({
  loginText: {
    color: colors.white,
    fontWeight: 'bold',
  },
  blackbutton: {
    marginTop: 40,
    marginBottom: 16,
    paddingTop: 12,
    paddingBottom: 12,
    paddingLeft: 38,
    paddingRight: 38,
    borderRadius: 2,
    elevation: 2,
    alignSelf: 'center',
    alignItems: 'center',
    backgroundColor: colors.darkGrey,
    justifyContent: 'center',
  },
  input: {
    height: 48,
    ios: {
      backgroundColor: colors.white,
      borderRadius: 4,
      padding: 8,
      marginBottom: 8,
      marginLeft: 16,
      marginRight: 16,
    },
  },
  forgotpass: {
    alignSelf: 'center',
    color: colors.darkGrey,
    marginBottom: 10,
    android: {
      fontFamily: 'sans-serif-medium',
    },
    ios: {
      fontFamily: 'System',
      fontWeight: '600',
    },
    fontSize: 14,
    height: 25,
  },
  loginstatus: {
    marginTop: 10,
    marginBottom: 10,
    height: 25,
    fontSize: 20,
    alignSelf: 'center',
    color: colors.white,
  },
  logo: {
    width: 260,
    height: 108,
    marginBottom: 48,
    resizeMode: 'contain',
    alignSelf: 'center',
  },
  wrapper: {
    paddingLeft: 16,
    paddingRight: 16,
    backgroundColor: colors.magenta,
    flex: 1,
    justifyContent: 'center',
  },
});

export default styles;
