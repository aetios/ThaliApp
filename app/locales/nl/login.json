{
  "Username": "Gebruikersnaam",
  "Password": "Wachtwoord",
  "LOGIN": "INLOGGEN",
  "Forgot password?": "Wachtwoord vergeten?"
}
