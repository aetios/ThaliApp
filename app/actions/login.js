import { AsyncStorage } from 'react-native';
import * as types from './actionTypes';

const USERNAMEKEY = '@MyStore:username';
const TOKENKEY = '@MyStore:token';

export function loginSuccess(username, token) {
  return {
    type: types.LOGINSUCCESS,
    loginState: 'loggedIn',
    username,
    token,
  };
}

export function loginProgress() {
  return {
    type: types.LOGINPROGRESS,
    loginState: 'progress',
  };
}

export function loginFailure() {
  return {
    type: types.LOGINFAILURE,
    loginState: 'failure',
  };
}

export function login(username, password) {
  return (dispatch) => {
    dispatch(loginProgress());
    const data = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        username,
        password,
      }),
    };
    return fetch('http://localhost:8000/api/token-auth/', data)
      .then(
        response => response.json())
      .then(
        (responseJson) => {
          if (responseJson.token) {
            return AsyncStorage.multiSet([[USERNAMEKEY, username], [TOKENKEY, responseJson.token]])
              .then(
                () => dispatch(loginSuccess(username, responseJson.token)),
              );
          }
          return dispatch(loginFailure());
        })
      .catch(() => dispatch(loginFailure()));
  };
}

export function logout() {
  AsyncStorage.multiRemove([USERNAMEKEY, TOKENKEY]);
  return {
    type: types.LOGOUT,
  };
}
