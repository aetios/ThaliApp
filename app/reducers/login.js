import * as types from '../actions/actionTypes';

const initialState = {
  loggedIn: false,
  loginError: false,
};

export default function login(state = initialState, action = {}) {
  switch (action.type) {
    case types.LOGIN:
      if (action.password === '42') {
        return { ...state, loginError: false, loggedIn: true };
      }
      return { ...state, loginError: true };
    default:
      return { ...state };
  }
}
