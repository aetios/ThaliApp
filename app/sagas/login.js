import { call, takeEvery, put } from 'redux-saga/effects';
import { AsyncStorage } from 'react-native';
import Snackbar from 'react-native-snackbar';

import { apiRequest } from '../utils/url';
import * as loginActions from '../actions/login';
import * as pushNotificationsActions from '../actions/pushNotifications';

export const USERNAMEKEY = '@MyStore:username';
export const TOKENKEY = '@MyStore:token';
export const DISPLAYNAMEKEY = '@MyStore:displayName';
export const PHOTOKEY = '@MyStore:photo';

const login = function* login(action) {
  const { user, pass } = action.payload;

  Snackbar.show({ title: 'Logging in', duration: Snackbar.LENGTH_INDEFINITE });

  const data = {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      username: user,
      password: pass,
    }),
  };
  try {
    const response = yield call(apiRequest, 'token-auth', data);
    const { token } = response;

    yield call(AsyncStorage.multiSet, [
      [USERNAMEKEY, user],
      [TOKENKEY, token],
    ]);
    yield put(loginActions.success(user, token));
    yield put(loginActions.profile(token));
    yield put(pushNotificationsActions.register());
    Snackbar.dismiss();
    Snackbar.show({ title: 'Login successful' });
  } catch (error) {
    Snackbar.dismiss();
    Snackbar.show({ title: 'Login failed' });
  }
};

const logout = function* logout() {
  yield call(AsyncStorage.multiRemove, [USERNAMEKEY, TOKENKEY]);
  yield put(pushNotificationsActions.invalidate());
  Snackbar.show({ title: 'Logout successful' });
};

const tokenInvalid = function* tokenInvalid() {
  yield call(AsyncStorage.clear);
  yield put(pushNotificationsActions.invalidate());
};

const profile = function* profile(action) {
  const { token } = action.payload;

  const data = {
    method: 'GET',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Token ${token}`,
    },
  };

  try {
    const userProfile = yield call(apiRequest, 'members/me', data);

    yield call(AsyncStorage.multiSet, [
      [DISPLAYNAMEKEY, userProfile.display_name],
      [PHOTOKEY, userProfile.avatar.medium],
    ]);
    yield put(loginActions.profileSuccess(userProfile.display_name, userProfile.avatar.medium));
  } catch (error) {
    // Swallow error
  }
};

const loginSaga = function* loginSaga() {
  yield takeEvery(loginActions.LOGIN, login);
  yield takeEvery(loginActions.LOGOUT, logout);
  yield takeEvery(loginActions.PROFILE, profile);
  yield takeEvery(loginActions.TOKEN_INVALID, tokenInvalid);
};

export default loginSaga;
