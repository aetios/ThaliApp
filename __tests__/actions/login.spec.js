import * as actions from '../../app/actions/login';

describe('login actions', () => {
  it('should expose the login actions', () => {
    expect(actions.SUCCESS).toEqual('LOGIN_SUCCESS');
    expect(actions.LOGIN).toEqual('LOGIN_LOGIN');
    expect(actions.LOGOUT).toEqual('LOGIN_LOGOUT');
    expect(actions.PROFILE).toEqual('LOGIN_PROFILE');
    expect(actions.PROFILE_SUCCESS).toEqual('LOGIN_PROFILE_SUCCESS');
  });

  it('should create an action to log the user in', () => {
    expect(actions.login('username', 'password')).toMatchSnapshot();
  });

  it('should create an action for a successful login', () => {
    expect(actions.success('username', 'token')).toMatchSnapshot();
  });

  it('should create an action to log the user out', () => {
    expect(actions.logout()).toMatchSnapshot();
  });

  it('should create an action to load the user profile', () => {
    expect(actions.profile('token')).toMatchSnapshot();
  });

  it('should create an action for a successful user profile load', () => {
    expect(actions.profileSuccess('displayName', 'photo')).toMatchSnapshot();
  });
});