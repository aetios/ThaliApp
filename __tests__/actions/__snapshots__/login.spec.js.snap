// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login actions should create an action for a successful login 1`] = `
Object {
  "payload": Object {
    "token": "token",
    "username": "username",
  },
  "type": "LOGIN_SUCCESS",
}
`;

exports[`login actions should create an action for a successful user profile load 1`] = `
Object {
  "payload": Object {
    "displayName": "displayName",
    "photo": "photo",
  },
  "type": "LOGIN_PROFILE_SUCCESS",
}
`;

exports[`login actions should create an action to load the user profile 1`] = `
Object {
  "payload": Object {
    "token": "token",
  },
  "type": "LOGIN_PROFILE",
}
`;

exports[`login actions should create an action to log the user in 1`] = `
Object {
  "payload": Object {
    "pass": "password",
    "user": "username",
  },
  "type": "LOGIN_LOGIN",
}
`;

exports[`login actions should create an action to log the user out 1`] = `
Object {
  "type": "LOGIN_LOGOUT",
}
`;
